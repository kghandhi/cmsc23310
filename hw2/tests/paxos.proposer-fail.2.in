# Example with two proposers.
# 
# Same as paxos.proposer-fail.1.in, but with P1 recovering.

2 3 50
0 PROPOSE 1 42
8 FAIL PROPOSER 1
11 PROPOSE 2 37
26 RECOVER PROPOSER 1
0 END
