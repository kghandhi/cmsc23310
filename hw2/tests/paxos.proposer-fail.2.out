  0:    -> P1  PROPOSE v=42
  1: P1 -> A1  PREPARE n=1
  2: P1 -> A2  PREPARE n=1
  3: P1 -> A3  PREPARE n=1
  4: A1 -> P1  PROMISE n=1 (Prior: None)
  5: A2 -> P1  PROMISE n=1 (Prior: None)
  6: A3 -> P1  PROMISE n=1 (Prior: None)
  7: P1 -> A1  ACCEPT n=1 v=42
  8: ** P1 FAILS **
  9:
 10:
 11:    -> P2  PROPOSE v=37
 12: P2 -> A1  PREPARE n=2
 13: P2 -> A2  PREPARE n=2
 14: P2 -> A3  PREPARE n=2
 15: A1 -> P2  PROMISE n=2 (Prior: n=1, v=42)
 16: A2 -> P2  PROMISE n=2 (Prior: None)
 17: A3 -> P2  PROMISE n=2 (Prior: None)
 18: P2 -> A1  ACCEPT n=2 v=42
 19: P2 -> A2  ACCEPT n=2 v=42
 20: P2 -> A3  ACCEPT n=2 v=42
 21: A1 -> P2  ACCEPTED n=2 v=42
 22: A2 -> P2  ACCEPTED n=2 v=42
 23: A3 -> P2  ACCEPTED n=2 v=42
 24:
 25:
 26: ** P1 RECOVERS **
 26: P1 -> A2  ACCEPT n=1 v=42
 27: P1 -> A3  ACCEPT n=1 v=42
 28: A1 -> P1  ACCEPTED n=1 v=42
 29: A2 -> P1  REJECTED n=1
 30: A3 -> P1  REJECTED n=1
 31: P1 -> A1  PREPARE n=3
 32: P1 -> A2  PREPARE n=3
 33: P1 -> A3  PREPARE n=3
 34: A1 -> P1  PROMISE n=3 (Prior: n=2, v=42)
 35: A2 -> P1  PROMISE n=3 (Prior: n=2, v=42)
 36: A3 -> P1  PROMISE n=3 (Prior: n=2, v=42)
 37: P1 -> A1  ACCEPT n=3 v=42
 38: P1 -> A2  ACCEPT n=3 v=42
 39: P1 -> A3  ACCEPT n=3 v=42
 40: A1 -> P1  ACCEPTED n=3 v=42
 41: A2 -> P1  ACCEPTED n=3 v=42
 42: A3 -> P1  ACCEPTED n=3 v=42

P1 has reached consensus (proposed 42, accepted 42)
P2 has reached consensus (proposed 37, accepted 42)
